content/posts/2021/01/2021-11-04-Getting-Started-With-React/index.md

```diff
@@ -0,0 +1,334 @@
+- children: An array of **HTML Elements** nested within a component.
+
+So the same component we just created can be written as:
+
+        const App = ()=>{
+            return(
+                React.createElement('div', null, React.createElement('h1',null, "Hello World"),
+                
+                React.createElement('div', null, React.createElement('h1',null, "This is our first react app, isn't it amazing!!!!"))
+            );
+         }
+
+This is a whole lot to type right? It looks messy. If you trace through it you will realize we are creating a `div` which has no props passed to it and has other elements created inside it which are `h1` and `h2`. If you've interacted with JavaScript a lot you'll realize that it a lot similar to ``document.createElement`` after all this is a JavaScript library.
+
+This is the beauty of JSX, it allows you to write JavaScript and HTML without going through the hustle of ``React.createElement()``.
+React developers almost exclusively use JSX, this section was important to understand what happens under the hood.
+
+
+### Making things dynamic
+
+Why use JSX when you can just write HTML and get away with it? Well if we remember what JSX stands for( JavaScript XML ). This means we can use JavaScript to make things dynamic. Our previous example looks like so:
+
+     const App = () =>{
+            return(
+                <div>
+                <h1>Our First React App</h1>
+                <h2>This is our first react app, isn't it amazing!!!!</h2>
+                </div>
+            )
+        }
+
+Now let's make it dynamic.
+Create a variable ``message`` to hold our message.
+
+``const message = 'This is our first react app, isn't it amazing!!!!';``
+
+Now to add JavaScript to use this, we use **curlybraces** `{}`.
+
+        const App = ()=>{
+            const message = 'This is our first react app, isn't it amazing!!!!';
+            return(
+                <h1>Our First React App</h1>
+                <h2>{ message }</h2>
+            )
+        }
+
+If you run this in your terminal you will see the output. Now go ahead and change the message and see magic!
+
+We use curly braces to tell the compiler **"execute this code in JavaScript"**. If we didn't have the curly braces the message wouldn't be executed as JavaScript, and instead the text 'message' would be displayed on your browser.
+
+## How a component get's rendered.
+
+Hopefully, I have cleared all the questions you have about JSX and creating components plus making them dynamic. I know you still asking yourself how these components get rendered right? This section will clear that up.
+Let's take a look at our file structure, notice ``index.html`` , this is always considered an entry file, to most developers if not all. You can change the entry point if you wish. For now we will leave it as is.
+
+Focus on this line of code
+> ReactDOM.render(<App />, document.getElementById("root"));
```

Comment:
This should not be a block-quote.

Format it as a code block 

![Screenshot 2021-11-28 at 23 27 11](https://user-images.githubusercontent.com/9129843/143784692-a9bae98f-eafb-4e2f-a21b-617892a7a5ef.png)